components/supervisor/pkg/terminal/terminal_test.go

```diff
@@ -49,15 +46,12 @@ func TestTerminals(t *testing.T) {
 				t.Fatal("no terminal")
 			}
 			stdoutOutput := bytes.NewBuffer(nil)
-			go io.Copy(stdoutOutput, terminal.Stdout.Listen())
-
-			for _, stdin := range test.Stdin {
-				terminal.PTY.Write([]byte(stdin + "\r\n"))
-				time.Sleep(test.Delay)
-			}
-
-			err = terminalService.Mux.Close(resp.Alias)
-			if err != nil {
+			go func() {
+				command := strings.Join(append(test.Stdin, "sleep 1", "exit\r\n"), " && ")
+				terminal.PTY.Write([]byte(command))
+			}()
+			_, err = io.Copy(stdoutOutput, terminal.Stdout.Listen())
```

Comment:
It seems there is some kind of race: 
```
    terminal_test.go:61: unexpected output (-want +got):
          []string{
        - 	`echo "yarn" && echo "gp sync-done init" && echo "yarn --cwd theia-training watch" && history && sleep 1 && exit`,
        - 	"",
          	"",
          }
```

writer was closed, before delivering the complete content